### PR TITLE
calibtool: switch to `libcalib` library

### PR DESCRIPTION
**Pull request by niewim19**
## Description

Recent addition of `libcalib` makes a lot of `calibttol` redundant and unnecessary. Changes in this PR make `calibtool` dependent on `libcalib`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: calib/magmot.c [hunk 1]
```diff
@@ -222,6 +191,7 @@ static int cal_magmotRun(void)
 	float thrtl = 0, thrtlStep, startThrtl = 0.3;
 	float x[CALIB_POINTS], y[3][CALIB_POINTS];
 	unsigned int m, pts;
+	float (*motorEq)[3][3];
```
[the remark this answers is not in the document]
Comment (niewim19) [reply calib/magmot.c:194]: I do not understand this github remark, I assume its wrong.

File: calib/calibtool.h [hunk 2]
```diff
@@ -30,12 +32,13 @@ typedef struct _calib_t {
 
 	/* utility related */
 	const char *(*help)(void);             /* help message description */
-	int (*interpret)(const char *, float); /* calibration file data interpreter */
 	int (*write)(FILE *);                  /* calibration file data write */
-} calib_t;
+
+	calib_t *(*calStructGet)(void); /* returns internal structure of libcalib:calib_t type */
+} calibration_t;
```
Comment (HBuczynski) [inline calib/calibtool.h:38]: You can call it `calib_ops_t`. `ops` means operations and the `calib_t` can be named `calib_data_t`

File: calib/calibtool.h [hunk 3]
```diff
@@ -30,12 +32,13 @@ typedef struct _calib_t {
 
 	/* utility related */
 	const char *(*help)(void);             /* help message description */
-	int (*interpret)(const char *, float); /* calibration file data interpreter */
 	int (*write)(FILE *);                  /* calibration file data write */
-} calib_t;
+
+	calib_t *(*calStructGet)(void); /* returns internal structure of libcalib:calib_t type */
```
Comment (HBuczynski) [inline calib/calibtool.h:37]: Can it return const value?
```suggestion
	calib_t *(*dataGet)(void); /* returns internal structure of libcalib:calib_t type */
```

File: calib/magiron.c [hunk 4]
```diff
@@ -19,61 +19,21 @@
...
-		case CHAR_HARDIRON:
-			mat = &magiron_common.softCal;
-			break;
-		default:
-			return NULL;
-	}
-
-	/* matrix boundary checks performed by matrix_at() */
-	return matrix_at(mat, row, col);
+	magiron_common.params.type = typeMagiron;
```
Comment (HBuczynski) [inline calib/magiron.c:35]: It should be set in `init` function

File: calib/magiron.c [hunk 5]
```diff
@@ -96,30 +56,15 @@ static void magiron_printIron(FILE *file, char type, matrix_t *mat)
 static int cal_magironWrite(FILE *file)
 {
 	/* Printing hard iron calibration parameters */
-	magiron_printIron(file, CHAR_HARDIRON, &magiron_common.hardCal);
+	magiron_printIron(file, CHAR_HARDIRON, &magiron_common.params.params.magiron.hardCal);
```
Comment (HBuczynski) [inline calib/magiron.c:59]: `params.params.magiron` does not look good.

File: calib/main.c [hunk 6]
```diff
@@ -17,7 +17,7 @@
 #include <stdbool.h>
 #include <errno.h>
 
-#include "calib.h"
+#include "calibtool.h"
 #include "hmap.h"
 
 #define CALIB_FILE  "/etc/calib.conf" /* Path to calibration parameters file */
```
Comment (HBuczynski) [inline calib/main.c:23]:
```suggestion
#define PATH_CALIB_FILE  "/etc/calib.conf" /* Path to calibration parameters file */
```

File: calib/magiron.c (hunk 4 above)
Comment (niewim19) [reply calib/magiron.c:35]: But we need it at read time which happens before `init`. Moved this to constructor.

Comment (HBuczynski) [reply calib/magiron.c:35]: Ok, constructor is a good place as well.

File: calib/magiron.c (hunk 5 above)
Comment (niewim19) [reply calib/magiron.c]: changed the common structure field name to `data`

**Review by HBuczynski — APPROVED** [review]

**Pull request merged**